Zend/zend_vm_def.h

```diff
@@ -5554,7 +5554,112 @@ ZEND_VM_C_LABEL(num_index):
 	ZEND_VM_NEXT_OPCODE_CHECK_EXCEPTION();
 }
 
-ZEND_VM_HANDLER(71, ZEND_INIT_ARRAY, CONST|TMP|VAR|CV, CONST|TMPVAR|UNUSED|NEXT|CV, ARRAY_INIT|REF)
+ZEND_VM_HANDLER(208, ZEND_ADD_ARRAY_UNPACK, ANY, ANY)
+{
+	USE_OPLINE
+	zend_free_op free_op1;
+	zval *op1;
+	
+	SAVE_OPLINE();
+	op1 = GET_OP1_ZVAL_PTR(BP_VAR_R);
+	
+	SEPARATE_ARRAY(EX_VAR(opline->result.var));
+ZEND_VM_C_LABEL(add_again):
+	if (EXPECTED(Z_TYPE_P(op1) == IS_ARRAY)) {
+		HashTable *ht = Z_ARRVAL_P(op1);
+		zval *val;
+		zend_string *key;
+
+		ZEND_HASH_FOREACH_STR_KEY_VAL(ht, key, val) {
+			if (key) {
+				zend_throw_error(NULL, "Cannot unpack array with string keys");
+				FREE_OP1();
+				HANDLE_EXCEPTION();
+			} else {
+				Z_TRY_ADDREF_P(val);
```

Comment:
You need to unwrap rc=1 references here (and test this).

Reply:
why (to unwrap reference here)?

Reply:
References with refcount 1 should behave like non-references in most circumstances.

Reply:
hmm. I checked and updated this PR (with rc=1 test), although I still don't understand why we do so.

thanks for your help and I'm now looking into the opcache part. thanks @nikic